packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -4,6 +4,320 @@
 
 import 'package:package_config/package_config.dart';
 
+const String _currentDirectoryScript = r"""
+var _currentDirectory = (function () {
```

Comment:
Looking into this more, it looks like the old logic was because google3 has implicit multi-roots at subdirectories. E.g., `blah:404/google3/` and `blah:404/workspace/`. We can replace this with the host root (`document.location.host`) if Flutter apps don't run in multi-app zones and are served from a single root.

I don't quite see how to pipe this in, though it's very likely I'm missing the right pointer.

Reply:
Closing the loop here, we're passing this root directory in from the build system, but it's defaulted to '/'. I believe this is the correct app root for all Flutter Web apps.